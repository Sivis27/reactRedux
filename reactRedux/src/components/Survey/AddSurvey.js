import React from 'react';
import { Form, Button } from 'react-bootstrap';

const AddSurvey = props => {
  return (
    <Form onSubmit={props.handleAddTodo}>
      <Form.Group controlId="item">
        <Form.Label>What you want to do?</Form.Label>
        <Form.Control type="text" name="todo" placeholder="Enter todo" />
      </Form.Group>
      <Button variant="primary" type="submit">
        Add Survey
      </Button>
    </Form>
  );
};

export default AddSurvey;