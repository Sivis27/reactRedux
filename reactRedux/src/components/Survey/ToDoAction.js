import React from 'react'

const ToDoAction = (props) => {
    return (
        <div className="todo-item">
            <h2>{props.items}</h2>
        </div>
    )
}

export default ToDoAction;