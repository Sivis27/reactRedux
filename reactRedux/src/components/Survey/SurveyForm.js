import React from 'react';
import ToDoAction from './ToDoAction';

const TodoLists = (props) => {
    return (
        <div className="todo-list">
            {props.items.length === 0 && <p>No Survey / Feedbacks found. Please add some.</p> }
            { props.items.map( (todo, index) => <ToDoAction key={index} items={todo}></ToDoAction>) }
        </div>
    );
};

export default TodoLists;