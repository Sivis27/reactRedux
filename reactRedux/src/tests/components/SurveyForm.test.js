import React from 'react';
import { shallow } from 'enzyme';
import Survey from '../../components/Survey/SurveyForm';
test('should test Survey component with default state of empty array', () => {
 const wrapper = shallow(<Survey items={[]} />);
 expect(wrapper).toMatchSnapshot();
}); 

test('should test SurveyForm component with list of objects', () => {
    const wrapper = shallow(<Survey items={['Apple','Mango']} />);
    expect(wrapper).toMatchSnapshot();
   });

