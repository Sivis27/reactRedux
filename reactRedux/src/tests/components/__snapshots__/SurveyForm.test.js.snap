// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should test Survey component with default state of empty array 1`] = `
<div
  className="todo-list"
>
  <p>
    No Survey / Feedbacks found. Please add some.
  </p>
</div>
`;

exports[`should test SurveyForm component with list of objects 1`] = `
<div
  className="todo-list"
>
  <ToDoAction
    items="Apple"
    key="0"
  />
  <ToDoAction
    items="Mango"
    key="1"
  />
</div>
`;
